app.py

```diff
@@ -60,11 +91,72 @@ def respond(
     ],
 )
 
-with gr.Blocks() as demo:
+# --- 3. BLOCKS STRUCTURE WITH TWO TABS (Chatbot and Robot) ---
+
+with gr.Blocks(title="Robot Control and Chatbot MCP") as demo:
     with gr.Sidebar():
         gr.LoginButton()
-    chatbot.render()
+    
+    # FIRST TAB: The original Chatbot
+    with gr.Tab("Chatbot"): 
+        chatbot.render()
+        
+    # SECOND TAB: The Robot Control / MCP Tools Interface
+    with gr.Tab("Robot Control MCP"): 
+        gr.Markdown("## 🤖 Robot Control and Telemetry Interface")
+        robot_output = gr.Textbox(label="Last Action Result / Telemetry")
+
+        # Telemetry
+        with gr.Row():
+            telemetry_btn = gr.Button("Update Telemetry", variant="secondary")
+            # MCP EXPOSURE: api_name="get_telemetry"
+            telemetry_btn.click(
+                fn=get_telemetry_sync, 
+                inputs=[], 
+                outputs=robot_output, 
+                api_name="get_telemetry"
+            )
+
+        # Movement Controls
+        with gr.Row():
+            distance_input = gr.Number(label="Forward Distance (meters)", value=1.0)
+            forward_btn = gr.Button("Move Forward", variant="primary")
+            # MCP EXPOSURE: api_name="move_forward"
+            forward_btn.click(
+                fn=move_forward_sync,
+                inputs=[distance_input],
+                outputs=robot_output,
+                api_name="move_forward" 
+            )
+        
+        with gr.Row():
+            left_btn = gr.Button("Turn Left (90°)")
+            right_btn = gr.Button("Turn Right (90°)")
+            
+            # MCP EXPOSURE: api_name="turn_left"
+            left_btn.click(
+                fn=turn_left_sync,
+                inputs=[],
+                outputs=robot_output,
+                api_name="turn_left" 
+            )
+            # MCP EXPOSURE: api_name="turn_right"
+            right_btn.click(
+                fn=turn_right_sync,
+                inputs=[],
+                outputs=robot_output,
+                api_name="turn_right" 
+            )
+
 
+# --- 4. STARTING THE MCP SERVER (Crucial) ---
 
 if __name__ == "__main__":
-    demo.launch()
+    print("Starting Gradio/MCP server...")
+    # The mcp_server=True option enables the exposure of API endpoints defined by api_name.
+    demo.launch(
+        mcp_server=True, 
```

Comment:
I noticed that you added the option

```text
mcp_server=True
```

But I could not find the tool descriptions here:

<img width="1503" height="789" alt="Image" src="https://github.com/user-attachments/assets/83c8592f-3f09-416f-a113-c543230c3743" />